src/DetailsView/components/failure-instance-panel-control.tsx

```diff
@@ -133,39 +150,73 @@ export class FailureInstancePanelControl extends React.Component<FailureInstance
     private getFailureInstancePanelDetails = (): JSX.Element => {
         return (
             <FailureInstancePanelDetails
-                path={this.state.path}
-                snippet={this.state.snippet}
+                path={this.state.currentInstance.path}
+                snippet={this.state.currentInstance.snippet}
                 onSelectorChange={this.onSelectorChange}
                 onValidateSelector={this.onValidateSelector}
             />
         );
     };
 
+    private getCurrentInstance = (): FailureInstanceData => {
+        const updatedInstance = {
+            failureDescription: this.state.currentInstance.failureDescription,
+            path: this.state.currentInstance.path,
+            snippet: this.state.currentInstance.snippet,
+        };
+
+        return updatedInstance;
+    };
+
     protected onFailureDescriptionChange = (event, value: string): void => {
-        this.setState({ failureDescription: value });
+        const updatedInstance = this.getCurrentInstance();
+        updatedInstance.failureDescription = value;
+        this.setState({ currentInstance: updatedInstance });
     };
 
     private onSelectorChange = (event, value: string): void => {
-        this.setState({ path: value });
+        const updatedInstance = this.getCurrentInstance();
+        updatedInstance.path = value;
+        this.setState({ currentInstance: updatedInstance });
     };
 
     private onValidateSelector = (event): void => {
-        const currSelector = this.state.path;
-        this.setState({ snippet: 'snippet for ' + currSelector });
+        const currSelector = this.state.currentInstance.path;
+        const currSnippet = 'snippet for ' + currSelector;
```

Comment:
I know this is from an earlier implementation and not this PR but we shouldn't be adding UI elements (like 'snippet for') in strings like this. Something we might want to change in a future PR.

Reply:
I put it in currently just to make sure it updated based on the selector when i clicked validate--is it okay to leave for testing purposes until the end?

Reply:
Since it's already in master, I'm fine with moving forward but it's something we should come back to sooner than later? 😄

Reply:
Sounds good! I'll try to get this cleaned up in the next PR so we don't leave it floating